app/models/metric.rb

```diff
@@ -1,3 +1,37 @@
+# Note on the database layout of `metrics` and `metrics_rollups`
+#
+# In the database, the `metrics` table leverages [PostgreSQL table inheritance]
+# and [PostgreSQL table partioning].
+#
+# Table inheritance is used to create child tables with the same structure as
+# the parent.  As columns are added or removed from the parent table, they are
+# automatically added and removed from the child tables.  The `metrics` table,
+# which is used for "realtime" data, has 24 child tables, one per hour.  The
+# `metrics_rollups` table, which is used for long term "rollup" data, has 12
+# child tables, one per month.
+#
+# Table partitioning is used to divert inserts and deletes from the parent table
+# to a specific child table, and to allow queries to search across the various
+# child tables and return the results as if they were in a single table.  This
+# is accomplished using triggers on the parent table, which look at the hour of
+# the timestamp for `metrics` or the month of the timestamp for
+# `metric_rollups`, and then diverting to the corresponding child table.
+# Additionally, a shared sequence is used from the parent table for the primary
+# key, so that when queries blend the rows together the primary key constraint
+# is satisfied.
+#
+# The reason for all of this is because of read/write contention.  The metrics
```

Comment:
> The reason for all of this is because of read/write contention.

This sharding helps reduce read/write contention.

Reply:
> 'We may want to try and avoid the word "this"'
> \- Keenan Brock